### ISSUE TITLE
Language switcher

### ISSUE THREAD

**Issue by peterlauri** [infoportugal/wagtail-modeltranslation#60]
We are trying out wagtail and wagtail-modeltranslation. It works fine using djangos i18n_patterns and all. However, we have one problem. We wanna be able to switch language, and it should switch to the same page (if it exist) in the swapped language.

How to map a page url to the same page url in the other language? It is not as simple as just replacing the /en/ with /sv/, as for example the slug could be different if one is using that. Any standard way to do this? Any helper methods?

**Comment by fdemmer** [comment]
I wrote the following template tag to get the url path of the current request, but in a different language; works for wagtail pages and normal django views:

```python
@register.simple_tag(takes_context=True)
def switch_language(context, requested_language=None, *args, **kwargs):
    """
    Get active page's url by a specified language
    Usage: {% switch_language 'en' %}
    """
    request = context['request']

    if request.resolver_match:
        # handle wagtail pages
        if request.resolver_match.url_name == 'wagtail_serve':
            with translation.override(requested_language):
                current_page = context.get('page')
                url = current_page.relative_url(request.site)

        # handle django views
        else:
            url_parts = resolve(request.path)
            with translation.override(requested_language):
                url = reverse(url_parts.view_name, kwargs=url_parts.kwargs)

        return force_text(url)
    return ''
```

Since we only have two languages, the switcher template snipped is pretty simple/stupid:
```python
{% if request.LANGUAGE_CODE == 'de' %}
    <a href="{% switch_language 'en' %}">English</a>
{% else %}
    <a href="{% switch_language 'de' %}">Deutsch</a>
{% endif %}
```

**Comment by SalahAdDin** [comment]
But these function shouldn't be extern, wagtail-modeltranslation should be usable with a simple language chooser.

**Comment by peterlauri** [comment]
@fdemmer thanks. We will try it out. Is there any corner cases you have bumped into when using this?

**Comment by fdemmer** [comment]
@peterlauri well, it's in use for a project, that is not yet live, but so far we haven't seen any problems (it's a single-site project, so with multiple sites in one app, there might be some more aspects to think about). 

the concept is pretty simple: find out what current view/page is, force langauge, generate url.

so as long as the view/page can be determined and `relative_url()`/`reverse()` work properly with the language override. this should be pretty solid.

if you can think of anything, please let us know :)

**Comment by dmarcelino** [comment]
This seems to be the use case for [change_lang](https://github.com/infoportugal/wagtail-modeltranslation/blob/master/wagtail_modeltranslation/templatetags/wagtail_modeltranslation.py#L24-L53) which is available since at least 0.6. Please reopen if that's not the case. Thanks
